File: app/src/main/java/com/ebsoft/watchlist/network/IEX/IEXApi.java
package com.ebsoft.watchlist.network.IEX;

import com.ebsoft.watchlist.data.model.IEX.Quote;
import com.ebsoft.watchlist.data.model.IEX.StockQuote;

import java.util.Map;

import io.reactivex.Observable;
import retrofit2.Response;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface IEXApi {

    @GET("1.0/stock/market/batch?&types=quote")
    Observable<Response<Map<String, StockQuote>>> getQuote(@Query("symbols") String symbols);
}

File: app/src/main/java/com/ebsoft/watchlist/data/control/network/IEX/IEXApi.kt
package com.ebsoft.watchlist.data.control.network.IEX

import com.ebsoft.watchlist.data.model.IEX.StockQuote
import com.ebsoft.watchlist.data.model.IEX.Symbol
import kotlinx.coroutines.Deferred
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Path

interface IEXApi {

    @GET("/stable/stock/{symbols}/batch?types=quote")
    fun getQuote(@Path("symbols") symbols: String): Deferred<Response<StockQuote>>

    @GET("/stable/search/{symbol}")
    fun searchSymbol(@Path("symbol") symbol: String): Deferred<Response<List<Symbol>>>
}
